src/wp-includes/functions.php

```diff
@@ -7830,6 +7830,28 @@ function wp_unique_id( $prefix = '' ) {
 	return $prefix . (string) ++$id_counter;
 }
 
+/**
+ * Generates an incremental ID that is independent per each different prefix.
+ *
+ * It is similar to `wp_unique_id`, but each prefix has its own internal ID
+ * counter to make each prefix independent from each other. The ID starts at 1
+ * and increments on each call. The returned value is not universally unique,
+ * but it is unique across the life of the PHP process and it's stable per
+ * prefix.
+ *
+ * @since 6.4
+ *
+ * @param string $prefix Optional. Prefix for the returned ID. Default empty string.
+ * @return string Incremental ID per prefix.
+ */
+function wp_unique_prefixed_id( $prefix = '' ) {
+	static $id_counters = array();
+	if ( ! array_key_exists( $prefix, $id_counters ) ) {
+		$id_counters[ $prefix ] = 0;
+	}
+	return $prefix . (string) ++$id_counters[ $prefix ];
```

Comment:
An empty line before `return` for consistency.

```suggestion

	return $prefix . (string) ++$id_counters[ $prefix ];
```